tests/slack_sdk/socket_mode/test_interactions_builtin.py

```diff
@@ -46,6 +48,16 @@ def test_buffer_size_validation(self):
     def test_interactions(self):
         if is_ci_unstable_test_skip_enabled():
             return
+
+        default_recursion_limit = sys.getrecursionlimit()  # will restore later
+        # This built-in WebSocket client internally has recursive method calls of _fetch_messages method.
```

Comment:
As mentioned in this comment, this test case uses a very small data buffer size and that can result in recursion error. If someone optimizes the internals to eliminate this, it would be a nice improvement. That being said, I believe that this is not a critical issue at this moment. The benefit of the current implementation is that the code is easier to read than the approach to avoid recursive method calls.